store/src/java/com/zimbra/cs/lmtpserver/ZimbraLmtpBackend.java

```diff
@@ -696,7 +696,7 @@ private void deliverMessageToLocalMailboxes(Blob blob, Blob blobEEW, BlobInputSt
                                 // Get msgid first, to avoid having to reopen and reparse the blob
                                 // file if Mailbox.addMessageInternal() closes it.
                                 pm.getMessageID();
-                                if (blobEEW != null && ExternalEmailWarning.getInstance().isEnabled()
+                                if (blobEEW != null && ExternalEmailWarning.getInstance().isEnabled(account)
```

Comment:
Check for ExternalEmailWarning.getInstance().isEnabled(account) stored in variable can be reused here and on #530